unity/Assets/Scripts/BaseFPSAgentController.cs

```diff
@@ -7275,6 +7223,110 @@ ProceduralAsset asset
             actionFinishedEmit(true, geoList);
         }
 
+        private Bounds GetAgentBounds(GameObject gameObject, Type agentType) {
+            Debug.Log(agentType);
+            Debug.Log(typeof(StretchAgentController));
+            Bounds bounds = new Bounds(gameObject.transform.position, Vector3.zero);
+            MeshRenderer[] meshRenderers = gameObject.GetComponentsInChildren<MeshRenderer>();
+            if (agentType == typeof(LocobotFPSAgentController)) {
+                meshRenderers = this.baseAgentComponent.BotVisCap.GetComponentsInChildren<MeshRenderer>();
+            } else if (agentType == typeof(StretchAgentController)) {
+                meshRenderers = this.baseAgentComponent.StretchVisCap.GetComponentsInChildren<MeshRenderer>();
+            }
+            foreach (MeshRenderer meshRenderer in meshRenderers) {
+                bounds.Encapsulate(meshRenderer.bounds);
+            }
+            return bounds;
+        }
+
+        public void SpawnBoxCollider(GameObject agent, Type agentType, Vector3 scaleRatio, bool useAbsoluteSize = false, bool useVisibleColliderBase = false) {
```

Comment:
Since this is now a helper function to help spawn the agent colliders, I'd change the name to be a bit more descripting and also lead with a lowercase so its not confused with an action that should be called directly

Something like `spawnAgentBoxCollider`